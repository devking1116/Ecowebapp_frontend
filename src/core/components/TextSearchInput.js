/* @flow */
import React from 'react';
import reactCSS from 'reactcss';
import { Input } from 'semantic-ui-react';

type Props = {

};

/**
 * The text input used to perform a string serach.
 */
export default function TextSearchInput({ ...otherProps }: Props): React.Element<*> {
  const styles = reactCSS({
    'default': {
      input: {
        width: '100%',
      },
    },
  });

  return (
    <div {...otherProps}>
      <Input style={styles.input} icon='search' placeholder='Search...' />
    </div>
  );
}
