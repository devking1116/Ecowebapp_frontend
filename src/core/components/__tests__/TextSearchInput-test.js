import React from 'react';
import renderer from 'react-test-renderer';

import TextSearchInput from '../TextSearchInput';

test('it displays the serach input.', () => {
  const component = renderer.create(
    <TextSearchInput />
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
