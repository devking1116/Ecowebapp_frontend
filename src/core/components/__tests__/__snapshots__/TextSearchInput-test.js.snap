exports[`test it displays the serach input. 1`] = `
<div>
  <div
    className="ui icon input"
    style={
      Object {
        "width": "100%",
      }
    }>
    <input
      placeholder="Search..."
      type="text" />
    <i
      aria-hidden="true"
      className="search icon" />
  </div>
</div>
`;
